pkg/instrumentation/helper.go

```diff
@@ -63,6 +69,59 @@ func isAutoInstrumentationInjected(pod corev1.Pod) bool {
 	return false
 }
 
+// Look for duplicates in the provided containers.
+func findDuplicatedContainers(ctrs []string) error {
+	// Merge is needed because of multiple containers can be provided for single instrumentation.
+	mergedContainers := strings.Join(ctrs, ",")
```

Comment:
I am sure if I understand the logic of this func. 

To find a duplicate sorting and then checking for two consecutive equal items should be enough.

Reply:
> To find a duplicate sorting and then checking for two consecutive equal items should be enough.

In this case it's not enough. It's possible that `ctrs` will look like this `[]string{"app1,app2", "app1", "app1,app3,app4", "app4"},` - concatenated slice of multiple strings coming from language specific container names annotations. We need to merge and then split container names to obtain slice of all provided names one by one.

Reply:
got it, thanks. Splitting each array item individually would work as well here.